webapp/channels/src/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx

```diff
@@ -21,74 +22,56 @@ export interface Props extends PropsFromRedux {
     currentTeamName: string;
 }
 
-export default class SidebarBaseChannel extends React.PureComponent<Props> {
-    handleLeavePublicChannel = (callback: () => void) => {
-        this.props.actions.leaveChannel(this.props.channel.id);
+const SidebarBaseChannel = ({
+    channel,
+    currentTeamName,
+    actions,
+}: Props) => {
+    const intl = useIntl();
+
+    const handleLeavePublicChannel = useCallback((callback: () => void) => {
+        actions.leaveChannel(channel.id);
         trackEvent('ui', 'ui_public_channel_x_button_clicked');
         callback();
-    };
+    }, [channel.id]);
```

Comment:
I don't expect that to ever change, but this also depends on `actions.leaveChannel`.